pkg/controller/elasticsearch/pdb/reconcile.go

```diff
@@ -116,12 +116,20 @@ func expectedPDB(es esv1beta1.Elasticsearch, statefulSets sset.StatefulSetList,
 }
 
 // buildPDBSpec returns a PDBSpec computed from the current StatefulSets,
-// considering the cluster health and topology.
+// considering the cluster health and topology, when cluster is high available.
+// it sets MinAvailable to 0 when nodecount is 1 (not high available)
+//   when the PDB is 1, and the amount of pods is 1, Kubernetes will not be able to reboot the node.
 func buildPDBSpec(es esv1beta1.Elasticsearch, statefulSets sset.StatefulSetList) v1beta1.PodDisruptionBudgetSpec {
-	// compute MinAvailable based on the maximum number of Pods we're supposed to have
+	// compute MinAvailable based on the maximum number of Pods we're supposed to have.
 	nodeCount := statefulSets.ExpectedNodeCount()
-	// maybe allow some Pods to be disrupted
-	minAvailable := nodeCount - allowedDisruptions(es, statefulSets)
+	var minAvailable int32
+	if nodeCount == 1 {
+		// when only single node, allow the pod to be rebooted, to ensure Kubernetes nodes do not get hung up by the pdb.
+		minAvailable = 0
+	} else {
+		// maybe allow some Pods to be disrupted
+		minAvailable = nodeCount - allowedDisruptions(es, statefulSets)
+	}
```

Comment:
I think it would make sense to have this block as part of the `allowedDisruptions` function where the logic around checking health and number of nodes exists.
This function returns the number of nodes that can safely be disrupted. In this situation I think it could check:

```go
if actualSsets.ExpectedNodeCount() == 1 {
    // single node cluster (not highly-available)
    // allow the node to be disrupted to ensure K8s nodes operations can be performed
    return 1
}
```

Also this function is fairly easy to unit test. I think it has an existing unit test that you probably want to modify:

```go
{
    name: "green health but single-node cluster: 0 disruption allowed",
    args: args{
        es:          esv1.Elasticsearch{Status: esv1.ElasticsearchStatus{Health: esv1.ElasticsearchGreenHealth}},
				actualSsets: sset.StatefulSetList{sset.TestSset{Replicas: 1, Master: true, Data: true}.Build()},
        },
        want: 0,
},
```

Reply:
Good point, in my haste I read over it.
I moved the logic there.

There still is the state that if someone creates a nodeset with a single node, that pod will still be blocked from rebooting. But perhaps that is to the user to fix.